lib/Contracts.js

```diff
@@ -4,61 +4,69 @@ const Web3 = require('web3');
-    let web3 = oThis.originWeb3;
-    return BTContracts.getBrandedToken(web3, address, options);
-  }
-
-  GatewayComposer(address, options) {
-    const oThis = this;
-    let web3 = oThis.originWeb3;
-    return BTContracts.getGatewayComposer(web3, address, options);
-  }
-
-  UtilityBrandedToken(address, options) {
-    const oThis = this;
-    let web3 = oThis.auxiliaryWeb3;
-    return BTContracts.getUtilityBrandedToken(web3, address, options);
-  }
-
-  static getBrandedToken(web3, address, options) {
-    web3 = Contracts._getWeb3(web3);
-    let jsonInterface = abProvider.getABI('BrandedToken');
-    let contract = new web3.eth.Contract(jsonInterface, address, options);
+  /**
+   * Returns BrandedToken contract instance.
+   *
+   * @param originWeb3 Origin chain web3 object.
+   * @param address BrandedToken contract address.
+   * @param options Tx options.
+   * @returns {web3.eth.Contract} Contract instance.
+   * @constructor
+   */
+  static getBrandedToken(originWeb3, address, options) {
+    originWeb3 = Contracts._getWeb3(originWeb3);
+    let jsonInterface = abiBinProvider.getABI('BrandedToken');
+    let contract = new originWeb3.eth.Contract(jsonInterface, address, options);
     return contract;
   }
 
-  static getGatewayComposer(web3, address, options) {
-    web3 = Contracts._getWeb3(web3);
-    let jsonInterface = abProvider.getABI('GatewayComposer');
-    let contract = new web3.eth.Contract(jsonInterface, address, options);
+  /**
+   * Returns GatewayComposer contract instance.
+   *
+   * @param originWeb3 Origin chain web3 object.
+   * @param address GatewayComposer contract instance.
+   * @param options Tx options.
+   * @returns {web3.eth.Contract} Contract instance.
+   * @constructor
+   */
+  static getGatewayComposer(originWeb3, address, options) {
+    originWeb3 = Contracts._getWeb3(originWeb3);
+    let jsonInterface = abiBinProvider.getABI('GatewayComposer');
+    let contract = new originWeb3.eth.Contract(jsonInterface, address, options);
     return contract;
   }
 
-  static getUtilityBrandedToken(web3, address, options) {
-    web3 = Contracts._getWeb3(web3);
-    let jsonInterface = abProvider.getABI('UtilityBrandedToken');
-    let contract = new web3.eth.Contract(jsonInterface, address, options);
+  /**
+   * Returns UtilityBrandedToken contract instance.
+   *
+   * @param auxiliaryWeb3 Auxiliary chain web3 object.
+   * @param address UtilityBrandedToken contract address.
+   * @param options Tx options.
+   * @returns {web3.eth.Contract} Contract instance.
+   * @constructor
+   */
+  static getUtilityBrandedToken(auxiliaryWeb3, address, options) {
+    auxiliaryWeb3 = Contracts._getWeb3(auxiliaryWeb3);
+    let jsonInterface = abiBinProvider.getABI('UtilityBrandedToken');
+    let contract = new auxiliaryWeb3.eth.Contract(jsonInterface, address, options);
     return contract;
   }
 
+  /**
+   * Returns web3 instance. If web3 is string url, it constructs web3 object from string.
```

Comment:
Suggestion :-  **If input param `web3` is string url, it constructs web3 object from it.**